proparse/src/main/java/org/prorefactor/proparse/antlr4/Lexer.java

```diff
@@ -780,12 +804,13 @@ ProToken id(int inputTokenType) throws IOException {
       }
     }
     // See if it's a keyword
-    if (ttype == ProParserTokenTypes.ID)
+    if (ttype == PreprocessorParser.ID)
       ttype = NodeTypes.testLiteralsTable(currText.toString(), ttype);
     return makeToken(ttype);
   }
 
   ProToken ampText() throws IOException {
+    LOGGER.trace("Entering ampText()");
     for (;;) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Reduce the total number of break and continue statements in this loop to use at most one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS135)